src/server/db/ModelHelpers/studentHelpers.js

```diff
@@ -52,7 +52,14 @@ exports.checkInStudents = async (req, res) => {
     const matches = await recognize(url);
     const date = await moment().format('YYYY-MM-DD hh:mm:ss');
     const [matchedUsers] = await Student.getMatchedUsers(matches);
-    await Student.checkInOnTime(matchedUsers, date);
+    for (let i = 0; i < matchedUsers.length; i++) {
+      let [cutOffDate] = await Student.getAttendanceStatus(matchedUsers[i].users_id, date.slice(0, 10))
+      if (cutOffDate[0].status === 'Pending') {
+        await Student.checkInOnTime(matchedUsers[i].users_id, date);
+      } else {
+        matchedUsers.splice(i, 1);
```

Comment:
not resetting index will skip over the next value,

ex [1, 2, 3, 4]
splice 2 when i is 1
[1, 3, 4]
3 is now index 1, but loop goes to index 2